<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LabFinal</class>
 <widget class="QMainWindow" name="LabFinal">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1375</width>
    <height>561</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonTextUnderIcon</enum>
  </property>
  <widget class="QWidget" name="widget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="0" column="0">
     <layout class="QGridLayout" name="window">
      <item row="0" column="1">
       <widget class="QGroupBox" name="output">
        <property name="minimumSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="title">
         <string>Output</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <widget class="QFrame" name="frame_output">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QGroupBox" name="source">
        <property name="minimumSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="title">
         <string>Source</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QFrame" name="frame_source">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QGroupBox" name="properties">
        <property name="minimumSize">
         <size>
          <width>250</width>
          <height>250</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>250</width>
          <height>389</height>
         </size>
        </property>
        <property name="title">
         <string>Properties</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="1" column="0">
          <widget class="QGroupBox" name="segmentation">
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="title">
            <string>Segmentation</string>
           </property>
           <widget class="QPushButton" name="pb_mot">
            <property name="geometry">
             <rect>
              <x>30</x>
              <y>50</y>
              <width>31</width>
              <height>31</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
           <widget class="QPushButton" name="pb_cvs">
            <property name="geometry">
             <rect>
              <x>90</x>
              <y>50</y>
              <width>31</width>
              <height>31</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
           <widget class="QPushButton" name="pb_ms">
            <property name="geometry">
             <rect>
              <x>150</x>
              <y>50</y>
              <width>31</width>
              <height>31</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QGroupBox" name="conversion">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="title">
            <string>Conversion</string>
           </property>
           <widget class="QPushButton" name="pb_grayscale">
            <property name="geometry">
             <rect>
              <x>60</x>
              <y>50</y>
              <width>31</width>
              <height>31</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
           <widget class="QPushButton" name="pb_hsv">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>50</y>
              <width>31</width>
              <height>31</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QGroupBox" name="edgeDetection">
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="title">
            <string>Edge Detection</string>
           </property>
           <widget class="QPushButton" name="pb_robets">
            <property name="geometry">
             <rect>
              <x>20</x>
              <y>50</y>
              <width>31</width>
              <height>31</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
           <widget class="QPushButton" name="pb_sobel">
            <property name="geometry">
             <rect>
              <x>70</x>
              <y>50</y>
              <width>31</width>
              <height>31</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
           <widget class="QPushButton" name="pb_scharr">
            <property name="geometry">
             <rect>
              <x>120</x>
              <y>50</y>
              <width>31</width>
              <height>31</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
           <widget class="QPushButton" name="pb_prewitt">
            <property name="geometry">
             <rect>
              <x>170</x>
              <y>50</y>
              <width>31</width>
              <height>31</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1375</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport_As">
     <property name="title">
      <string>Export As</string>
     </property>
     <addaction name="actionExportSource"/>
     <addaction name="actionExportOutput"/>
    </widget>
    <addaction name="actionOpenSource"/>
    <addaction name="actionSaveOutput"/>
    <addaction name="actionSaveAsOutput"/>
    <addaction name="menuExport_As"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <widget class="QMenu" name="menuClear">
     <property name="title">
      <string>Clear</string>
     </property>
     <addaction name="actionClearSource"/>
     <addaction name="actionClearOutput"/>
    </widget>
    <addaction name="menuClear"/>
    <addaction name="actionUndoOutput"/>
    <addaction name="actionRedoOutput"/>
   </widget>
   <widget class="QMenu" name="menuConversion">
    <property name="title">
     <string>Conversion</string>
    </property>
    <addaction name="actionRGBtoGrayscale"/>
    <addaction name="actionRGBtoHSV"/>
   </widget>
   <widget class="QMenu" name="menuSegmentation">
    <property name="title">
     <string>Segmentation</string>
    </property>
    <addaction name="actionMultiOtsuThresholding"/>
    <addaction name="actionChanVeseSegmentation"/>
    <addaction name="actionMorphologicalSnakes"/>
   </widget>
   <widget class="QMenu" name="menuEdge_Detection">
    <property name="title">
     <string>Edge Detection</string>
    </property>
    <addaction name="actionRoberts"/>
    <addaction name="actionSobel"/>
    <addaction name="actionScharr"/>
    <addaction name="actionPrewitt"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuConversion"/>
   <addaction name="menuSegmentation"/>
   <addaction name="menuEdge_Detection"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpenSource"/>
   <addaction name="actionSaveOutput"/>
   <addaction name="actionSaveAsOutput"/>
   <addaction name="actionUndoOutput"/>
   <addaction name="actionRedoOutput"/>
  </widget>
  <action name="actionOpenSource">
   <property name="text">
    <string>&amp;Open Source</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSaveOutput">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Save Output</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAsOutput">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Save As Output</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string>Shift+F4</string>
   </property>
  </action>
  <action name="actionExportSource">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Source</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionExportOutput">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Output</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+E</string>
   </property>
  </action>
  <action name="actionClearSource">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Source</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionClearOutput">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Output</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+L</string>
   </property>
  </action>
  <action name="actionUndoOutput">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Undo Output</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedoOutput">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Redo Output</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionRGBtoGrayscale">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;RGB to Grayscale</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="actionRGBtoHSV">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;RGB to HSV</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="actionMultiOtsuThresholding">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Multi-Otsu Thresholding</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="actionChanVeseSegmentation">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Chan-Vese Segmentation</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+C</string>
   </property>
  </action>
  <action name="actionMorphologicalSnakes">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Morphological Snakes</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+M</string>
   </property>
  </action>
  <action name="actionRoberts">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Roberts</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionSobel">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Sobel</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionScharr">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Scharr</string>
   </property>
   <property name="shortcut">
    <string>Shift+S</string>
   </property>
  </action>
  <action name="actionPrewitt">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Prewitt</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
